kr8s/_objects.py

```diff
@@ -965,43 +965,43 @@ def delete(  # type: ignore[override]
             propagation_policy=propagation_policy,
             grace_period=grace_period,
             force=force,
-        )  # type: ignore
+        )
 
     def refresh(self) -> None:  # type: ignore[override]
-        return run_sync(self.async_refresh)()  # type: ignore
+        return run_sync(self.async_refresh)()
 
     def patch(self, patch, *, subresource=None, type=None) -> None:  # type: ignore[override]
-        return run_sync(self.async_patch)(patch, subresource=subresource, type=type)  # type: ignore
+        return run_sync(self.async_patch)(patch, subresource=subresource, type=type)
 
     def scale(self, replicas=None) -> None:  # type: ignore[override]
-        return run_sync(self.async_scale)(replicas=replicas)  # type: ignore
+        return run_sync(self.async_scale)(replicas=replicas)
 
     def watch(self) -> Generator[tuple[str, Self]]:  # type: ignore[override]
-        yield from run_sync(self.async_watch)()
+        yield from run_sync_gen(self.async_watch)()
 
     def wait(  # type: ignore[override]
         self,
         conditions: list[str] | str,
         mode: Literal["any", "all"] = "any",
         timeout: int | float | None = None,
     ) -> None:
-        return run_sync(self.async_wait)(conditions, mode=mode, timeout=timeout)  # type: ignore
+        return run_sync(self.async_wait)(conditions, mode=mode, timeout=timeout)
 
     def annotate(self, annotations=None, **kwargs) -> None:  # type: ignore[override]
-        return run_sync(self.async_annotate)(annotations, **kwargs)  # type: ignore
+        return run_sync(self.async_annotate)(annotations, **kwargs)
 
-    def label(self, labels=None, **kwargs) -> None:  # type: ignore[override]
-        return run_sync(self.async_label)(labels, **kwargs)  # type: ignore
+    def label(self, *labels: dict | str, **kwargs) -> None:  # type: ignore[override]
```

Comment:
This seems like a change in behaviour no? Before `labels` was a kwarg, now it unpacks unnamed positionals into a `list[dict|str]`.

Reply:
Yeah ok I see this is a bug in the sync mixin. Thanks for picking this up.